src/EditorFeatures/Core/EditorConfigSettings/Data/IEditorConfigSettingInfo.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.EditorConfigSettings.Data
+{
+    internal interface IEditorConfigSettingInfo
+    {
+        string? GetSettingName();
+
+        string? GetDocumentation();
+
+        string[]? GetSettingValues(OptionSet optionSet);
+    }
```

Comment:
document this type and tehse methods.  also, explain under what circumstance you get null back from these methods.,